### ISSUE TITLE
[BUG] Spark-3.4 integration test failures due to code update in FileFormatWriter.

### ISSUE THREAD

**Issue by nartal1** [NVIDIA/spark-rapids#7144]
**Describe the bug**
Below integration tests are failing:

```
FAILED ../../src/main/python/hive_write_test.py::test_optimized_hive_bucketed_fallback[PARQUET-[Byte, Short, Integer, Long, Float, Double, String, Boolean, Date, Timestamp, Decimal(7,3), Decimal(12,2), Decimal(20,2)]][ALLOW_NON_GPU(DataWritingCommandExec)] - pyspark.sql.utils.IllegalArgumentException: Part of the plan is not columnar class org.apache.spark.sql.execution.SortExec
FAILED ../../src/main/python/hive_write_test.py::test_optimized_hive_bucketed_fallback[ORC-[Byte, Short, Integer, Long, Float, Double, String, Boolean, Date, Timestamp, Decimal(7,3), Decimal(12,2), Decimal(20,2)]][ALLOW_NON_GPU(DataWritingCommandExec)] - pyspark.sql.utils.IllegalArgumentException: Part of the plan is not columnar class org.apache.spark.sql.execution.SortExec
FAILED ../../src/main/python/orc_write_test.py::test_part_write_round_trip[String][IGNORE_ORDER] - pyspark.sql.utils.IllegalArgumentException: Part of the plan is not columnar class org.apache.spark.sql.execution.ProjectExec
FAILED ../../src/main/python/parquet_write_test.py::test_part_write_round_trip[String][IGNORE_ORDER] - pyspark.sql.utils.IllegalArgumentException: Part of the plan is not columnar class org.apache.spark.sql.execution.ProjectExec

```

**Steps/Code to reproduce bug**
Please provide a list of steps or a code sample to reproduce the issue.
```
./run_pyspark_from_build.sh -k test_optimized_hive_bucketed_fallback // first 2 failures
./run_pyspark_from_build.sh -k test_part_write_round_trip // last 2 failures
```

**Expected behavior**
Above tests should pass.

**Environment details (please complete the following information)**
 - Environment location: [Standalone, YARN, Kubernetes, Cloud(specify cloud provider)]
 - Spark configuration settings related to the issue

**Additional context**

**Comment by nartal1** [comment]
Test failures are due to https://github.com/NVIDIA/spark-rapids/issues/6074 . We should decide if we need to update tests to make it to pass or should we include SortOrder as mentioned in the issue for Spark-3.4.  In the short term, we can update tests.

**Comment by jlowe** [comment]
Fixed by #7167
